Update bluetooth documentation for bleak 0.19 changes (#1497)
home-assistant/developers.home-assistant: docs/bluetooth.md

### Fetching the bleak `BLEDevice` from the `address`

> Integrations should avoid the overhead of starting an additional scanner to resolve the address by calling the `bluetooth.async_ble_device_from_address` API, which returns a `BLEDevice` if any configured `bluetooth` scanner can reach the device.

Author: ```suggestion
Integrations should avoid the overhead of starting an additional scanner to resolve the address by calling the `bluetooth.async_ble_device_from_address` API, which returns a `BLEDevice` for the nearest configured `bluetooth` scanner that can reach the device. If no scanners can reach the device, the `bluetooth.async_ble_device_from_address` API, will return `None`.
```

> Integrations should avoid the overhead of starting an additional scanner to resolve the address may call the `bluetooth.async_ble_device_from_address` API, which returns a `BLEDevice` if the `bluetooth` integration scanner has recently seen the device. The integration MUST fall back to connecting via the `address` if the `BLEDevice` is unavailable.

Reviewer: This doesn't sound right:
> Integrations should avoid the overhead of starting an additional scanner to resolve the address may call the

Author: Also the note that they MUST fall back is wrong now as well.  If it doesn't return the device they won't be able to reach it no matter how hard they try as the underlying stack can't see it

Author: Adjusted

> Integrations should avoid the overhead of starting an additional scanner to resolve the address may call the `bluetooth.async_ble_device_from_address` API, which returns a `BLEDevice` if the `bluetooth` integration scanner has recently seen the device. The integration MUST fall back to connecting via the `address` if the `BLEDevice` is unavailable.

Author: ```suggestion
Integrations should avoid the overhead of starting an additional scanner to resolve the address by calling the `bluetooth.async_ble_device_from_address` API, which returns a `BLEDevice` if any configured `bluetooth` scanner can reach the device.
```

ble_device = bluetooth.async_ble_device_from_address(hass, "44:44:33:11:23:42", connectable=True)
```

> ### Fetching the bleak `BluetoothServiceInfoBleak` for a device

Author: ```suggestion
### Fetching the latest `BluetoothServiceInfoBleak` for a device
```
